core/src/main/java/com/google/instrumentation/stats/MeasurementDescriptor.java

```diff
@@ -34,13 +34,22 @@
    * Constructs a new {@link MeasurementDescriptor}.
    */
   public static MeasurementDescriptor create(
+      MeasurementDescriptor.Name name, String description, MeasurementUnit unit) {
+    return new AutoValue_MeasurementDescriptor(name.asString(), description, unit);
+  }
+
+  /**
+   * Constructs a new {@link MeasurementDescriptor}.
+   */
+  public static MeasurementDescriptor create(
       String name, String description, MeasurementUnit unit) {
     return new AutoValue_MeasurementDescriptor(StringUtil.sanitize(name), description, unit);
   }
 
   /**
    * Name of measurement, e.g. rpc_latency, cpu. Must be unique.
    */
+  // TODO(sebright): Change the type of this field to MeasurementDescriptor.Name.
```

Comment:
I think it will be hard to just change this return type. Maybe better if we add a new method that returns the MeasurementDescriptor.Name and this will just return the name.asString.